contracts/rdt/RevenueDistributionToken.sol

```diff
@@ -210,6 +211,17 @@ contract RevenueDistributionToken is IRevenueDistributionToken, ERC20 {
         _mint(shares_ = triBalanceUnstaked, triBalanceUnstaked, receiver_, address(this));
     }
 
+    function claimAndStake(
+        address masterChef_,
+        address receiver_,
+        uint256 pid_
+    ) external {
+        uint256 claimedRevenueAssets_ = _claim(receiver_);
+
+        ERC20(revenueAsset).approve(masterChef_, claimedRevenueAssets_);
+        IMasterChefV2(masterChef_).deposit(pid_, claimedRevenueAssets_, receiver_);
+    }
```

Comment:
Does this need an `Event`?

Reply:
events are from the external contract calls